src/mozanalysis/experiment.py

```diff
@@ -399,6 +440,8 @@ def build_metrics_query(
             time_limits (TimeLimits): An object describing the
                 interval(s) to query
             enrollments_table (str): The name of the enrollments table
+            exposure_based (bool): Use exposures as basis for calculating
+                metrics if True, otherwise use enrollments.
```

Comment:
Type theorists have started to convince me that using enums or a permitted set of strings is better than bools because they give you more flexibility but I don't have a really strong opinion.

Is it possible to use this interface interactively? I think we don't expect notebook users to call this method. The notebook API is here: https://mozilla.github.io/mozanalysis/api/experiment.html

Also, this suggests that we aren't taking the tack of representing these like segments, though thinking about that I suppose that's fair -- that doesn't actually make any sense because we need some day to describe the enrolled and exposed _values_ separately and segments don't permit that mechanism. More to think about in Jetstream.

Reply:
Agreed. I added an `AnalysisBasis` enum. Not sure if there is a better name for this?

Reply:
sgtm!